events/README_ClientVPN.md

```diff
@@ -0,0 +1,59 @@
+# Sample Function
+
+The following is a sample Lambda function that receives a Client VPN connection handler request as an input and then validates the IP address input and checks whether the connection source IP is on the allowed list defined as a map inside the function. If the source IP matches an allowed IP address it allows the access, otherwise an error message is presented to the user. Debug logs are generated to CloudWatch Logs. (Note that by default anything written to Console will be logged as CloudWatch Logs events.)
+
+```go
+import (
+	"fmt"
+	"log"
+	"net"
+
+	"encoding/json"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+var (
+    AllowedIPs = map[string]bool{
+		"10.11.12.13": true,
+	}
+)
+
+func handler(request events.ClientVPNConnectionHandlerRequest) (events.ClientVPNConnectionHandlerResponse, error) {
+	requestJson, _ := json.MarshalIndent(request, "", "  ")
+	log.Printf("REQUEST: %s", requestJson)
+
+	sourceIP := request.PublicIP
+	if net.ParseIP(sourceIP) == nil {
+		return events.ClientVPNConnectionHandlerResponse{}, fmt.Errorf("Invalid parameter PublicIP passed in request: '%s'", sourceIP)
+	}
+
+	log.Printf("SOURCE IP: %s", sourceIP)
+
+	postureCompliance := []string{}
+
+	allowed, ok := AllowedIPs[sourceIP]
+	if !ok {
+		allowed = false
+		postureCompliance = []string{"BlockedSourceIP"}
+	}
+
+	if allowed {
+		log.Printf("Allowing access from: %s", sourceIP)
+	} else {
+		log.Printf("Blocking access from: %s", sourceIP)
+	}
+
+	return events.ClientVPNConnectionHandlerResponse{
+		Allow: allowed,
+		ErrorMsgOnFailedPostureCompliance: "You're accessing from a blocked IP range.",
```

Comment:
I'm not sure; the "You're accessing from a blocked IP range" is more likely to resonate with the user's point of view as he can successfully connect from another IP address.

"Your IP address is not allowed" sounds more like we authenticate only based on the IP address.

If I switch this to allow it would be something like "You're not accessing from an allowed IP range" which I think is more confusing than the original with the double-negation. What do you think?

Reply:
I'm not going to block on it so whatever you want is fine. From my point of view this is an example, we are likely not going to change it later to expand the list of allowed ips or to add more complicated denial logic. It currently allows/denies based on the client's IP only so messaging that reflects that makes sense to me.

Reply:
I've adjusted the message as well to make it clearer with what we're doing; I hope.